src/main/java/emu/grasscutter/game/home/GameHome.java

```diff
@@ -102,20 +105,24 @@ public boolean addUnlockedHomeBgm(int homeBgmId) {
         return true;
     }
 
-    public Set<Integer> getUnlockedHomeBgmListInfo() {
+    public Set<Integer> getUnlockedHomeBgmList() {
         if (this.unlockedHomeBgmList == null) {
             this.unlockedHomeBgmList = new HashSet<>();
-            addAllDefaultUnlockedBgmIds(this.unlockedHomeBgmList);
+        }
+
+        var set = getDefaultUnlockedHomeBgmIds();
+        if (!this.unlockedHomeBgmList.containsAll(set)) {
+            this.unlockedHomeBgmList.addAll(set);
```

Comment:
```suggestion
        if (this.unlockedHomeBgmList.addAll(getDefaultUnlockedHomeBgmIds())) {
```

`Set::addAll` returns true if the set changes at all from the operation